docs/proposals/diamond-design.md

```diff
@@ -0,0 +1,458 @@
+
+### Report about diamond-originated information on a bundle
+```
+datamon bundle get --repo {repo} --bundle {bundle-id} --format `{{ range .Diamonds}}{{ printf "%v" .}}{{ end }}`
+```
+or
+```
+datamon bundle get --repo {repo} --bundle {bundle-id} --with-diamonds
+```
+
+### Inspect conflicts
+
+```
+datamon bundle list files --repo {repo} --bundle {bundle-id} --with-conflicts|--with-checkpoints
+```
+
+This list files which experienced conflicts (resp. overwrites) during a diamond worklow (resp. an incremental update).
+
+```
+datamon bundle download --repo {repo} --bundle {bundle-id} --with-conflicts|--with-checkpoints --destination {local path}
+```
+
+This download the different conflicting (resp. checkpointed) versions of conflicting (resp. overwritten) files ({filename}-{split-id}).
+(filtering options apply like in `datamon bundle download`, so does the `datamon bundle download file` subcommand).
+
+Each downloaded file will be: either the original (winning) version, or suffixed split ID
+
+Example:
+```
+/destination/my-conflicting-file.data                                     #  <-- winner
+/destination/my-conflicting-file.data-0ujsszwN8NRY24YaXiTIE2VWDTS         #  <-- candidate
+/destination/my-conflicting-file.data-0ujsswThIGTUYm2K8FjOOfXtY1K         #  <-- candidate
+/destination/my-conflicting-file.data-0ujssxh0cECutqzMgbtXSGnjorm         #  <-- candidate
+```
+
+### Resolve conflicts
+
+If the system's rule did not select the appropriate version, a user may want to create a new
+bundle with the proper version (all versions are uploaded anyway)
+
+Using `bundle download` and `bundle download --with-conflicts` allows for inspecting and selecting things.
+A `bundle upload` may then be baked with the "merged" dataset.
+However this incurs a lot of data shuffling around for a few metatada fixup.
+
+A less resource intensive way is to proceed as follows:
+```
+datamon bundle conflicts resolve --bundle {bundle-id} --files {file-list} [--message {...} --label {...}]
+```
+The file list contains the new selected winner. In the example above, if we want the 3rd candidate to be selected, the file list is:
+```
+my-conflicting-file.data-0ujssxh0cECutqzMgbtXSGnjorm
+```
+
+This creates a new bundle with the specified conflicts resolved (the `-{split-id}` suffix is trimmed from the resulting file).
+Resolved conflicts are removed from the conflicts list in the new bundle. When there is no remaining conflict, the conflicts metadata are wiped out.
+
+
+If everything is fine and you just want to make it clear for all consumers of this bundle, you may force all conflicts to be resolved at once.
+```
+datamon bundle conflicts resolve --bundle {bundle-id} --resolve-all [--message {...} --label {...}]
+```
+This creates a new bundle with the same data (no upload here) and conflicts metadata are wiped out.
+The bundle on which conflicts where initially detected remains untouched for traceability.
+
+### Cancel a diamond
```

Comment:
Not in MVP

Reply:
ok put that one (cancel) as nice to have.